src/models/Account.js

```diff
@@ -2,137 +2,166 @@ import {
   EntityModel,
   BillingContact,
   ShippingContact,
-  Instrument,
-} from './index'
+  Instrument
+} from "./index";
 
-import { logError, logWarning } from '../helpers/Logger'
+import { logError, logWarning } from "../helpers/Logger";
 
-const ACCOUNTS_LIST_RESOURCE = '/v1/accounts'
+const ACCOUNTS_LIST_RESOURCE = "/v1/accounts";
 
 export class Account extends EntityModel {
-  accountId = ""
-  name = ""
-  email = ""
-  billingContact = new BillingContact()
-  shippingContact = new ShippingContact()
-  instrument = new Instrument()
+  accountId = "";
+  name = "";
+  email = "";
+  billingContact = new BillingContact();
+  shippingContact = new ShippingContact();
+  instrument = new Instrument();
 
   constructor(params = {}) {
-    super(params)
+    super(params);
     Object.keys(params).map(attrName =>
       this._setAttr(attrName, params[attrName])
-    )
+    );
     this.billingContact = new BillingContact(
-      !!params === true ? {
-        ...this.billingContact,
-        ...params.billingContact,
-      }
+      !!params === true
+        ? {
+            ...this.billingContact,
+            ...params.billingContact
+          }
         : {}
-    )
+    );
     this.shippingContact = new ShippingContact(
-      !!params === true ? {
-        ...this.shippingContact,
-        ...params.shippingContact,
-      } : {}
-    )
+      !!params === true
+        ? {
+            ...this.shippingContact,
+            ...params.shippingContact
+          }
+        : {}
+    );
     this.instrument = new Instrument(
-      !!params === true ? {
-        ...this.instrument,
-        ...params.instrument,
-      } : {}
-    )
+      !!params === true
+        ? {
+            ...this.instrument,
+            ...params.instrument
+          }
+        : {}
+    );
   }
 
   saveWithSecureForm(
     apiKey,
     form,
-    {
-      onError,
-      onComplete,
-      onNext,
-      onValidationError,
-    },
-    apiOptions,
+    { onError, onComplete, onNext, onValidationError },
+    apiOptions
   ) {
-    const { loggingLevel = "" } = apiOptions
+    const { loggingLevel = "" } = apiOptions;
 
     if (!!apiKey === false) {
-      throw new Error("Unable to call save. Empty `apiKey`, make sure you have set your publicKey prop.")
+      throw new Error(
+        "Unable to call save. Empty `apiKey`, make sure you have set your publicKey prop."
+      );
     }
-    if (apiKey.startsWith('sk_') === true) {
-      throw new Error("Unable to call save. You are attempting to use a secret key.")
+    if (apiKey.startsWith("sk_") === true) {
+      throw new Error(
+        "Unable to call save. You are attempting to use a secret key."
+      );
     }
```

Comment:
not sure if this doesn't mean a lot of changes in the codebase, but can we have these being rejected inside the `new Promise(...)` block?
The goal is to have this method always returning a promise, what do you think?

Reply:
We are going to leave this as a hard error since it is a configuration issue.  We will be moving the rest of validation and error handling into the Promise block and returning the same error/validation errors in `reject`

Reply:
got it